### PR TITLE
feat: use selectors for swapperApi state

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

The `isLoading` property of an RTK query hook only returns the state of the instantiation in the current context.
As an RTK query can be called from anywhere in the app, we need to have a global way of knowing if we are waiting for a response from an endpoint.

By using the API's reducer we can query the global state of the API as a whole, or an individual endpoint.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A - swapper improvements.

## Risk

Minimal - only affects loading state UI.

## Testing

When we are getting a quote or rates, the swapper should show appropriate loading states.

The main improvement here is that we should no longer see the "Rate not available" show before getting a rate, as we now correctly understand when we are searching for rates.

In addition, trade quotes refresh when the trade amounts change, so an "Updating quote..." state is now shown.

### Engineering

As above.

### Operations

As above.

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2936** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2936" target="_blank">Graphite</a> 
    * **PR #2937** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2937" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2937?utm_source=stack-comment).

File: src/state/apis/swapper/selectors.ts
Comment (0xApotheosis) [inline src/state/apis/swapper/selectors.ts:14]: The crux of this PR.

File: src/components/Trade/hooks/useTradeAmounts.tsx
Comment (0xApotheosis) [inline src/components/Trade/hooks/useTradeAmounts.tsx:211]: Moved to ensure updates are atomic.

File: src/components/Trade/TradeInputV2.tsx
Comment (0xApotheosis) [inline src/components/Trade/TradeInputV2.tsx:94]: If `true`, we don't need to show a skeleton over the input fields as we have enough information to estimate all values.

The quote may be stale, but "Preview Trade" is blocked until it refreshes.

File: src/state/apis/swapper/selectors.ts
Comment (gomesalexandre) [reply src/state/apis/swapper/selectors.ts:14]: LGTM, confirmed RTK doesn't have any notion of an aggregated global queries status of sort and this is the way to go
https://stackoverflow.com/questions/71523851/app-wide-rtk-query-loading-fetching-indicator

**Review by gomesalexandre — COMMENTED** [review]
Conceptual stamp pending runtime regression test, LGTM!

**Review by gomesalexandre — APPROVED** [review]
`Updating Quote` is correctly shown on input value change:


https://user-images.githubusercontent.com/17035424/193560130-b55e90e1-eecb-4407-af35-80f5816e4994.mov

Loading states are also now more reliable, using the minimum available amount of data to display a quote, tested on slow 3G network throttling to validate this implementation

🍗

**Pull request merged**
